core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3204,6 +3205,22 @@ private[spark] object Utils extends Logging {
     }
     files.toSeq
   }
+
+  /**
+   * Get the value of -XX:G1HeapRegionSize if we are using G1 GC,
+   * otherwise just return None
+   */
+  val G1HeapRegionSize: Option[Long] = {
+    Try {
+      val diagnostic = ManagementFactoryHelper.getDiagnosticMXBean()
```

Comment:
@WangGuangxin This is not Java11 and Java17 API.

Reply:
Great findings! We need to test this on other JVM versions too.

Reply:
@dongjoon-hyun Thanks for your findings. I've updated and checked against JDK8, 11, and 17.  also cc @attilapiros